docs/src/main/asciidoc/rest.adoc

```diff
@@ -2271,6 +2271,13 @@ By default, methods annotated with `@ServerExceptionMapper` do **not** run CDI i
 Users however can opt into interceptors by adding the corresponding annotations to the method.
 ====
 
+[TIP]
+====
+When mapping an exception to a `@ServerExceptionMapper` method, the cause of the exception normally does not come into play.
+However, in cases where the root cause of an issue is wrapped in some generic exception, the `@UnwrapException` comes in handy
+for essentially using the cause when resolving the exception mapper.
```

Comment:
I found this a bit confusing, because I know `cause` refers to `Exception.getCause()` but it's not entirely clear whether this is conveyed to the user, or we're talking about "cause" as an abstract concern, such as "well it's caused by a bug".
I found the use of `root cause` further confusing because, the "root" makes me think equally of the "root exception" (the wrapper) and the "real cause" (the wrapped exception).

So, perhaps something like this would be less confusing to people like me?

```
Some exception types in Java only serve as wrappers for other exceptions. Often, checked exceptions are wrapped into `RuntimeException` just to not have them declared in method `throws` parameters. Working with `CompletionStage` for example, will require `CompletionException`. There are many such exception types that are just wrappers around the real cause of the exception.

If you wish to make sure your exception mapper is called for your exception type even when it is wrapped by one of those wrapper exceptions, you can use `@UnwrapException` on the exception wrapper type:

public class MyExceptionWrapper extends RuntimeException {
 public MyExceptionWrapper(Exception cause) {
  super(cause);
 }
}

If you don't control that exceptoin wrapper type, you can place the annotation anywhere and specify the exception wrapper types it applies to as annotation parameter:

@UnwrapException({CompletionException.class, RuntimeException.class})
@ServerExceptionMapper
public Response handleMyException(MyException x) {
 // ...
}
```

Something like this?

Reply:
Sure yeah, I'll update the PR soon!

Reply:
Much better, thanks for the suggestion!